connect/runtime/src/main/java/org/apache/kafka/connect/cli/ConnectDistributed.java

```diff
@@ -106,6 +108,7 @@ public Connect startConnect(Map<String, String> workerProps) {
         // Create the admin client to be shared by all backing stores.
         Map<String, Object> adminProps = new HashMap<>(config.originals());
         ConnectUtils.addMetricsContextProperties(adminProps, config, kafkaClusterId);
+        adminProps.putIfAbsent(CLIENT_ID_CONFIG, "connect-cluster-" + config.groupId());
```

Comment:
Yeah, that's fair. I was thinking it might be useful broker-side to call this out as a Connect application but it's not a must-have and the stuttering that you've noted with `connect-cluster-connect-cluster` is not great. Will remove.

Reply:
One thing that I'm wondering a bit more about is if we should add some kind of unique identifier for each worker within the Connect cluster, since without one, it becomes harder to use broker logs to debug issues.

It's tempting to use the worker's advertised URL to identify it since that comes from the config and is likely to be human-readable. It may be a little ugly in some edge cases, though. Other possibilities are to use a hash of the advertised URL, or a UUID. Thoughts?

Reply:
Yeah the advertised URL is the only identifier we have for a unique worker but it could be "ugly" and pretty long. 

Ideally users would provide a unique identifier (like the pod name) as the client.id and we would prefix/suffix it with the groupId and a short name like `shared-admin`.

Reply:
Agreed with the behavior when users explicitly supply a client ID 👍

I'd still like to give different client IDs to workers in the other case if possible. I gave base 64-encoding the advertised URL a shot and it doesn't seem too ugly.

Now when a `client.id` is supplied in the worker config, the Kafka clients for, e.g., the config backing store will have the ID `<groupId>-<userSuppliedClientId>-configs`, and when one isn't supplied, the ID will be `<groupId>-<base64EncodedAdvertisedUrl>-configs`.

How does that look?

Reply:
Base64 can contain equal signs, I wonder if that could mess with metric names. I'm seriously considering doing nothing special if `client.id` is not there.

Reply:
Hmm... yeah, that's fair. I'd still like to do _something_ if the user doesn't provide a `client.id`; could we just leave out that portion and do something like `<groupId>-configs`? Tweaked the PR to use that logic; LMKWYT